[linux-6.6.y] i2c: smbus: Add support for Zhaoxin SMBUS controller

leoliu-oc opened the pull request:
zhaoxin inclusion
category: feature

-------------------

The Zhaoxin platform implements the SMBUS controller on the hardware, enabling information exchange and collaboration between devices using the SMBus protocol.

opsiff commented on drivers/i2c/busses/Kconfig:
```diff
@@ -348,6 +348,17 @@ config I2C_SCMI
+config I2C_ZHAOXIN_SMB
+	tristate "Zhaoxin SMBus Interface"
+	depends on PCI || COMPILE_TEST
+	default m
```
非X86的话也需要 default m么?

leoliu-oc replied:
新版中添加了对CPU_SUP_CENTAUR/CPU_SUP_ZHAOXIN的依赖。以保证只有在兆芯平台上，才会编译为module。

leoliu-oc committed:
i2c: smbus: Add support for Zhaoxin SMBUS controller

zhaoxin inclusion
category: feature

-------------------

The Zhaoxin platform implements the SMBUS controller on the hardware,
enabling information exchange and collaboration between devices using the
SMBus protocol.

Signed-off-by: leoliu-oc <leoliu-oc@zhaoxin.com>

opsiff commented:
/ok-to-test
